react/more-react-concepts/managing-state-with-context-api/managing-state-with-context-api.md

```diff
@@ -0,0 +1,259 @@
+### Introduction
+
+Throughout the lessons, we've learned how to manage state and how to pass data/props around components. However, as our application grow in size, not only will this process of managing state and passing data around becomes repetitive and inconvenient, it will also be hard to manage. This lesson will cover how we can reduce these complexities in our application by using The Context API.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Occurence of Prop Drilling in larger applications
+- Using the Context API to pass state and data deep into React components
+- Real-world application of the Context API
+- Drawbacks of using the Context API
+
+### What is Context API
+
+Context API in React is used commonly as a [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection). In React however, this is used as a way to pass data/functions deep into other components without the need of passing props to each of these components. That sounds pretty amazing doesn't it? It alleviates the pain points you may or may not have experienced in the previous projects. Let's take a look at an example of a project that we've built in the past:
```

Comment:
> Context API in React is used commonly as a [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection)

Interesting. I don't quite understand this statement. How exactly does Context API relate to dependency injection?

Reply:
As the name implies, we're basically injecting these dependencies regardless of where they are in the component tree. I guess it's a form of it? We can probably remove this info since it's also not really necessary to include